src/plugins/kibana/server/lib/case_conversion.js

```diff
@@ -0,0 +1,15 @@
+const _ = require('lodash');
+
+module.exports = {
+  convertToSnakeCase: function (object) {
+    return _.mapKeys(object, (value, key) => {
+      return _.snakeCase(key);
+    });
+  },
+
+  convertToCamelCase: function (object) {
```

Comment:
It was originally shallow because I didn't want it modifying ES specific names in the mappings. That's not an issue now, but I figured once someone needed it they would add a `convertToSnakeCaseDeep` method to this module.